bin/halcyon-build

```diff
@@ -0,0 +1,29 @@
+#!/bin/sh
+set -e
+
+private_storage_setup() {
+  [ -n "$HALCYON_AWS_ACCESS_KEY_ID" ] &&
+  [ -n "$HALCYON_AWS_SECRET_ACCESS_KEY" ] &&
+  [ -n "$HALCYON_S3_BUCKET" ]
+}
+
+[ -f ./.env ] && source ./.env
+
+if ! private_storage_setup; then
+  echo 'You must set the Halcyon variables for AWS credentials' >&2
+  echo 'https://halcyon.sh/reference/#private-storage-options' >&2
+  exit 1
+fi
+
+if [ ! -d /app/halcyon ]; then
+  echo 'Installing halcyon...'
+  git clone https://github.com/mietek/halcyon.git /app/halcyon
+fi
+
+# Note --no-clean-private-storage ensures developers don't need to rebuild their
+# stuff if another developer builds other versions for their own use (by
+# default, halcyon keeps only one version per platform at a time).
+/app/halcyon/halcyon build \
+  --no-clean-private-storage --cabal-version=1.22.0.0 "$@"
+
+ln -sf /app/sandbox/.halcyon-sandbox.config cabal.sandbox.config
```

Comment:
OK. In any case, you can always force Halcyon to rebuild the sandbox from scratch by using the [`HALCYON_SANDBOX_REBUILD`](https://halcyon.sh/reference/#halcyon_sandbox_rebuild) option.